src/lib.rs

```diff
@@ -578,17 +463,26 @@ fn rekey_entry(entry: &RagenixRule, identities: &[Box<dyn age::Identity>]) -> Re
 /// to the passed writer
 #[allow(clippy::missing_errors_doc)]
 #[allow(clippy::missing_panics_doc)]
-pub fn run<I, T>(itr: I, mut writer: impl Write) -> Result<()>
+pub fn run<I, T>(itr: I, mut writer: impl Write, mut writer_err: impl Write) -> Result<()>
 where
     I: IntoIterator<Item = T>,
     T: Into<OsString> + Clone,
 {
-    let opts = parse_args(itr);
+    let opts = cli::parse_args(itr);
 
     if opts.schema {
         write!(writer, "{}", AGENIX_JSON_SCHEMA)?;
     } else {
-        let rules = parse_rules(&opts.rules).unwrap();
+        if let Err(report) = validate_rules_file(&opts.rules) {
+            writeln!(
+                writer_err,
+                "error: secrets rules are invalid: '{}'\n{}",
+                &opts.rules, report
+            )?;
+            process::exit(1);
```

Comment:
Shouldn't it rather return an `Err` result, given that the signature uses a `Result` plus that it lives in lib?

Reply:
I thought about that as well but I wanted to prevent a detailed error/panic report which looks more like something unexpected happened. Actually, I regard this validation step a part of the command line parsing which also terminates the process if an invocation is invalid:

https://github.com/yaxitech/ragenix/blob/86c09b77f953253ab383bc02d4955252bb68a123/src/lib.rs#L471

Still, I agree that the filename `lib.rs` is rather misleading as it's not really intended to have someone use this code as library. I mostly organized the code like that to facilitate testing but ultimately decided to use `assert_cmd` anyways.